awswrapper/s3.go

```diff
@@ -334,6 +353,49 @@ func (o *S3Service) UploadToS3Concurrently(content []byte, bucketName string, pa
 	return err
 }
 
+func (o *S3Service) UploadToS3WithSession(session *ServiceSession, reader io.Reader, bucket, path string, contentType *string, options *S3Options, args ...interface{}) (err error) {
```

Comment:
What is the `UploadToS3WithSession` used for? We already have `UploadToS3Concurrently`, right? How is this different?

Reply:
This is to pass the session from outside. So that session will not be created everytime. I made a new function to keep backward compatibility.

Reply:
If both functions are the same except for passing the session, I think we should have some common function for the shared code.

Reply:
this is done